src/pal/tools/gen-buildsys.sh

```diff
@@ -30,6 +30,15 @@ then
   exit 1
 fi
 
+__CoreClrDir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )/../../.."
```

Comment:
This script gets __ProjectRoot from the build.sh as the first argument. It seems it has the same value as the __CoreClrDir here, so you should be able to use it.
Hmm, I can actually see that you've replaced the $1 with the __RepoRootDir, so I am not sure what was the reason for that.

Reply:
Based on my understanding the first argument is not always the CoreCLR or repo root - according to the code comment it's the <path to top level CMakeLists.txt> and gets called with different values. (The $1 argument is used at the very bottom of the script as the argument to the -S option for cmake.) Orthogonal to that I'm introducing the "standard" repo detection logic to identify __RepoRootDir to use as the base for the cross folder. For __ProjectRoot vs. __CoreClrRoot, over time I have come to the conclusion that __ProjectRoot (or __ProjectDir) most often refers to the CoreCLR subrepo root and naming it as such would be more descriptive but I feel somewhat reluctant to churn the codebase just to satisfy my naming whims.

Reply:
Based on our offline chat I have moved the tryrun.cmake script to the CoreCLR subrepo root.